src/jdk.dynalink/share/classes/jdk/dynalink/SecureLookupSupplier.java

```diff
@@ -39,13 +39,6 @@
  * @since 9
  */
 public class SecureLookupSupplier {
-    /**
-     * The name of a runtime permission required to successfully invoke the
-     * {@link #getLookup()} method.
-     */
-    public static final String GET_LOOKUP_PERMISSION_NAME = "dynalink.getLookup";
```

Comment:
This public member is [referenced from Nashorn](https://github.com/search?q=repo%3Aopenjdk%2Fnashorn%20GET_LOOKUP_PERMISSION_NAME&type=code). If we remove it, Nashorn will have to attempt to look it up reflectively as long as it supports older versions of Java. (Presuming the `java.security.AccessController` and related classes aren't also removed altogether. Are they?)

FWIW, as long as we're making breaking changes, this whole class could be removed – all it did was serve as a secure gate to access to a `MethodHandles.Lookup`. References to it could be replaced by the lookup object itself. I can understand if this might be too much work for this PR, though.

On the other hand, if we do _not_ remove the whole class, I'd prefer to keep the public string constant for the permission name so that we don't break binary compatibility for whoever might be referencing it (specifically, Nashorn.)

Reply:
Agreed, it is cleaner to retain the permission names in the public API.